api/src/main/java/org/motechproject/nms/api/web/BaseController.java

```diff
@@ -191,6 +192,20 @@ protected State getStateForFrontLineWorker(FrontLineWorker flw, Circle circle) {
         State state = frontLineWorkerService.getState(flw);
 
         if (state == null && circle != null) {
+            state = getStateFromCircle(circle);
+        }
+
+        return state;
+    }
+
+
+
+
+    protected State getStateFromCircle(Circle circle) {
+
+        State state = null;
+
+        if (circle != null) {
             List<State> states = circle.getStates();
 
             if (states.size() == 1) {
```

Comment:
Clarified from Saurabh that we need not to do the validation check if we are not sure about the state.
Thus, I have closed that bug NMS-205 but updated the code accordingly.